content/docs/reference/design-docs/signaling.en.md

```diff
@@ -0,0 +1,419 @@
+
+Signals are modeled as an evaluation function, taking a view of the world and returning the signal state
+
+```kotlin
+
+enum ZoneStatus {
+   /** The zone is clear to be used by the train */
+   CLEAR,
+   /** The zone is occupied by another train, but otherwise clear to use */
+   OCCUPIED,
+   /** The zone is incompatible. There may be another train as well */
+   INCOMPATIBLE,
+}
+
+interface MAView {
+    /** Combined status of the zones protected by the current signal */
+    val protectedZoneStatus: ZoneStatus
+    val nextSignalState: SignalState
+    val nextSignalSettings: SignalSettings
+}
+
+interface DirectSpeedLimit {
+    /** Distance between the signal and the speed limit */
+    val distance: Distance
+    val speed: Speed
+}
+
+interface IndirectSpeedLimit {
+    val distanceToNextSignal: Distance
+    val nextSignalState: SignalState
+    val nextSignalSettings: SignalSettings
+}
+
+interface SpeedLimitView {
+    /** A list of speed limits directly downstream of the signal */
+    val directSpeedLimits: List<DirectSpeedLimit>
+    /** A list of speed limits which need to be announced in a signal chain */
+    val indirectSpeedLimits: List<IndirectSpeedLimit>
+}
+
+fun signal(maView: MAView?, limitView: SpeedLimitView?): SignalState {
+    // ...
+}
+```
+
+The view should allow access to the following data:
+ - a synthetized view of zones downstream until the end of the train's MA
+ - the block chain
+ - the state of downstream signals which belong to the current block chain
+
+### Signaling view path
+
+The path along which the MAView and SpeedLimitView live is best expressed using blocks:
+
+- blocks can be added to extend the view along the path of a train
+- the view can be reduced by removing blocks, as the train passes by signals
+
+### Simulation outside the train path
+
+It is possible to simulate signaling outside of any train path:
+
+   - if a signal starts blocks which have differing paths, it is simulated as if it were at the end of a route
+   - if a signal starts blocks which all start the same path, it is simulated in the same view as the next signals in this path
+
+## Dependencies
+
+For the block graph generation:
+
+   - route graph. For each route:
+       - `waypoints: List<DiDetector>`
+       - `signals: OrderedMap<Position, UnloadedSignal>`
+       - `speed_limits: RangeMap<Position, SpeedLimit>`, including the logic for train category limits
+   - signaling systems
+   - drivers
+
+For evaluation:
+
+   - train path in blocks
+   - portion of the path to evaluate
+   - drivers
+   - state of the zones in the section to evaluate
+    
+## Operations
+
+   - **Instantiating a view** creates a framework for observing signals
+   - **Planning the path signals** to the view the blocks that the train will traverse
+   - **Observing a signal** subscribe to the state of a signal (through the view)
+   - **Passing a signal** signals that a signal has been passed by the train (through the view)
```

Comment:
```suggestion
   - **Instantiating a view** creates a framework for observing signals;
   - **Planning the path signals** to the view the blocks that the train will traverse;
   - **Observing a signal** subscribe to the state of a signal (through the view);
   - **Passing a signal** signals that a signal has been passed by the train (through the view).
```

Reply:
I beg to disagree. Most technical documentation, and even wikipedia, uses either regular dots or nothing at all.
https://en.wikipedia.org/wiki/Wikipedia:Manual_of_Style/Lists